pages/index.js

```diff
@@ -66,7 +66,9 @@ const Home = () => {
   const [showModal, setShowModal] = useState(false)
 
   const onSubmitURLs = useCallback(({ urls }) => {
-    setUrls([...urls])
+    setUrls(urls.map((url) => {
+      return {"url": new URL(url.url).toString()}
+    }))
```

Comment:
@hellais I added code to implement the punycode suggestion. I didn't do this where you suggested to do so (assuming I understand correctly your suggestion) because it seems yup.transform runs before validate (see https://github.com/jquense/yup/issues/242, also assuming I understand that issue correctly).

That said, I don't think we should do this transformation here. I think we should accept an URL containing unicode characters, which are JSON serializable, and then we should handle this case in the probe. We have tests in ooni/probe-cli checking whether we're able to handle i18n URLs, so the engine should be fine. I am not sure whether the mobile apps are able to handle an i18n URL, but, if the app does not handle an i18n URL, then this strikes me as an app bug more than anything else.

Reply:
Did you test the performance of this? In particular I would check to see if it's going to be responsive when copy pasting large URL lists into the form.

Re: the topic of puny coding of URLs, how does the engine write the final output to the measurement file? Will it set the input field of the measurement to have the URL encoded in puny code or will it use the unicode encoding of it? My concern is also related to how the data processing pipeline or third party consumers of the data will handle this case. For example it might be that the database field we are using for storing the input field is not unicode aware and this could potentially lead to problems.

Reply:
> Did you test the performance of this? In particular I would check to see if it's going to be responsive when copy pasting large URL lists into the form.

No, I did not. Doing this analysis seems something we should do if we keep this code. Thanks for suggesting it.

> Re: the topic of puny coding of URLs, how does the engine write the final output to the measurement file? [...]

Very insightful and important comment, thank you. For now, I can only provide you with this partial answer: I have investigated both Web Connectivity and Websteps behavior with an i18n input. The TL;DR is that they have both troubles to handle the i18n domain name and cannot properly measure.

Please, see my investigation here: https://github.com/ooni/probe/issues/1925.

To me, this seems part of a more comprehensive bug where we figure out how to completely handle such domain names and the first step seems to fix the CLI to make sure it can handle them properly (not as in "the units within the CLI know how to deal with this" but as in "the whole CLI successfully handle them").

I think it's a bit premature to sketch out a solution for this problem in this PR. I suggest we open an issue to acknowledge we have this broader i18n problem to solve and we need to investigate how the whole software pipeline deals with an i18n URL coming from OONI Run and how each component handles it.

Do you agree we should open a broader issue for this topic? I think allowing people to provide URLs with a path is an easier fix and should not block on fixing i18n for all our products, but I think this review has been super fruitful because it pinpointed a place in which we're not doing quite well.

Reply:
> Do you agree we should open a broader issue for this topic? 

Yes I agree with this. Based on your analysis it seems like this topic is much more complex and far reaching than anticipated. I think we should go ahead with landing this improvement to OONI Run and then make the relevant improvements to the rest of the stack.